src/theory/sep/theory_sep.cpp

```diff
@@ -69,6 +69,21 @@ TheorySep::~TheorySep() {
   }
 }
 
+void TheorySep::declareSepHeap(TypeNode locT, TypeNode dataT)
+{
+  if (!d_type_ref.isNull())
+  {
+    TypeNode te1 = d_loc_to_data_type.begin()->first;
+    std::stringstream ss;
+    ss << "ERROR: declaring heap constraints for two different types : ";
+    ss << locT << " -> " << dataT << ", but we already have ";
+    ss << d_type_ref << " -> " << d_type_data;
```

Comment:
This error message seems confusing to me. Either we only allow a single declare-heap command (then the message should say so) or we allow multiple declare-heap commands if the types are the same (then we should check this and not print an error)
The tests below seem to imply that we can only have a single declare-heap command...

Reply:
Yeah, you're right. I'll make this clearer.